src/main/java/net/dv8tion/jda/internal/entities/PermissionOverrideImpl.java

```diff
@@ -94,26 +101,45 @@ public long getDeniedRaw()
     @Override
     public JDA getJDA()
     {
-        return getChannel().getJDA();
+        return api.get();
     }
 
     @Override
     public Member getMember()
     {
-        return isMemberOverride() ? (Member) permissionHolder : null;
+        return getGuild().getMemberById(holderId);
     }
 
     @Override
     public Role getRole()
     {
-        return isRoleOverride() ? (Role) permissionHolder : null;
+        return getGuild().getRoleById(holderId);
     }
 
     @Nonnull
     @Override
     public GuildChannel getChannel()
     {
-        return channel.get();
+        JDAImpl jda = api.get();
+        GuildChannel channel;
+        switch (channelType)
+        {
+            case TEXT:
+                channel = jda.getTextChannelById(channelId);
+                break;
+            case VOICE:
+                channel = jda.getVoiceChannelById(channelId);
+                break;
+            case CATEGORY:
+                channel = jda.getCategoryById(channelId);
+                break;
+                //TODO: STORE
```

Comment:
Maybe we should add a `Guild#getChannelById(id): GuildChannel` for these things.